src/inputs/pipe.c

```diff
@@ -302,6 +305,42 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_end)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (errno != 0 && airplay_volume == 0) || (value_begin == value_end))
```

Comment:
Is errno guaranteed to be set by strtof if the conversion is successful? If not, then the second part may be undefined.